Remove firefox/nautilus from mandatory packages + use options.sh to install browser/filemanager post-install

redsh4de opened the pull request:
Instead of installing firefox & nautilus and setting them as defaults, this change will make it so the browser & file manager can be chosen by the user after installation of ML4W. This accounts for users that may already have browsers installed that they want to use, or those who do not wish to use a browser at all.

Instead of adding extra files/bloat to the codebase, this solution uses a slightly modified version of options.sh to allow targeted installs of additional packages without going through the initial menu

Changes made:
- Allow directly passing which package to modify for nl4w-options
- Removes firefox & nautilus from default packages
- Edited browser.sh and filemanager.sh to point to nl4w-options with the additional parameter by default
- Changes to all options files to ensure backwards compatibility
- Modified the waybar links, changed firefox icon to universal "web" icon

Commit:
change: use options.sh to install browser/filemanager

ArekKubinski commented:
Is there a way to retain firefox icon on waybar if the web browser is firefox? I really like my firefox icon and would pain me if I had to hack together a replace string script in post.sh

redsh4de commented:
> Is there a way to retain firefox icon on waybar if the web browser is firefox? I really like my firefox icon and would pain me if I had to hack together a replace string script in post.sh

Can definitely be done, i'll replace the default browser icon with the firefox icon if it is the chosen option in the options wizard.

Commit:
update: automatically update browser icon/name based on the browser installed
- allowed Ctrl+C to exit the scripts when launched directory

redsh4de commented:
@ArekKubinski This will now automatically update both the tooltip and icon whenever the browser is switched via ML4W options. (Currently the default font seems to only support firefox and chrome icons though)

mylinuxforwork commented:
@redsh4de With the latest push, firefox and nautilus will be only installed with an initial installation. I will look into your improvements.

redsh4de committed:
Merge branch 'main' into main

redsh4de committed:
Merge branch 'main' into main

redsh4de committed:
Merge branch 'main' into main

redsh4de committed:
Merge branch 'main' into main

redsh4de committed:
Merge branch 'main' into main

redsh4de committed:
remove 'break' statements from case

mylinuxforwork commented:
@redsh4de Thanks for your updates. I have just pushed an more readable version of the quicklinks.json to the main branch.

I will look into your options updates later after the release of 2.9.8

Commit:
updates
- add e-mail client to options menu
- show (none) in defaults when no email client is defined in the setting script

redsh4de commented:
@mylinuxforwork Gotcha! Yea the script will still work all the same, as it would look for the parent JSON object containing browser.sh

I also noticed that now email is included in the waybar, so i added e-mail installation options to the options.sh script + packages for both arch and fedora

mylinuxforwork commented:
@redsh4de Not able to test it before the release of 2.9.8.0 I will check again for 2.9.8.1. Hope that's Ok for you.

redsh4de commented:
@mylinuxforwork Sure thing!

Commit:
updates
- changed the fallback browser icon to the up-to-date one
- fixed a bug for selecting the file manager
- set options/email.sh to be the email client option installer by default

Commit:
newlines

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
add nautilus new window fix

Commit:
fix

redsh4de commented:
@mylinuxforwork Any opportunity to test the updates?
